src/storm/mesos/logviewer/SocketUrlDetection.java

```diff
@@ -0,0 +1,66 @@
+package storm.mesos.logviewer;
+
+import java.net.InetAddress;
+import java.net.Socket;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Optional;
+
+public class SocketUrlDetection implements IUrlDetection {
+
+    public static final int DEFAULT_PORT = 8000;
+    
+    public SocketUrlDetection() {
+        setPort(Optional.of(DEFAULT_PORT));
+    }
+    
+    public SocketUrlDetection(int port) {
+        setPort(Optional.of(port));
+    }
+    
+    public SocketUrlDetection(Optional<Integer> port) {
+        setPort(port);
+    }
+    
+    @Override
+    public boolean isReachable() {
+
+        Socket socket       = null;
+        boolean reachable   = false;
+        try {
+            
+            if (LOG.isDebugEnabled())
+                LOG.debug("Checking host " + InetAddress.getLocalHost() + " and port " + getPort() );
+            
+            if (port.isPresent()) {
+                socket      = new Socket(InetAddress.getLocalHost(), getPort().get());
```

Comment:
I guess I will never understand the phobia with spacing and newlines. I
have always felt it added readability to the code....but if I must I will
remove them.

On Tue, Jun 9, 2015 at 12:41 PM, Timothy Chen <notifications@github.com>
wrote:

> In src/storm/mesos/logviewer/SocketUrlDetection.java
> <https://github.com/mesos/storm/pull/38#discussion_r32034420>:
>
> > +    public SocketUrlDetection(Optional<Integer> port) {
> > +        setPort(port);
> > +    }
> > +
> > +    @Override
> > +    public boolean isReachable() {
> > +
> > +        Socket socket       = null;
> > +        boolean reachable   = false;
> > +        try {
> > +
> > +            if (LOG.isDebugEnabled())
> > +                LOG.debug("Checking host " + InetAddress.getLocalHost() + " and port " + getPort() );
> > +
> > +            if (port.isPresent()) {
> > +                socket      = new Socket(InetAddress.getLocalHost(), getPort().get());
>
> Spacing
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/storm/pull/38/files#r32034420>.
>